### ISSUE TITLE
Wiki's upgrade guide is not enough

### ISSUE THREAD

**Issue by falsandtru** [zfsonlinux/zfs#6689]
Wiki says as follows:

```
$ sudo yum remove zfs zfs-kmod spl spl-kmod libzfs2 libnvpair1 libuutil1 libzpool2 zfs-release
$ sudo yum install http://download.zfsonlinux.org/epel/zfs-release.el7_4.noarch.rpm
$ sudo yum install zfs 
```

https://github.com/zfsonlinux/zfs/wiki/RHEL-%26-CentOS#rhelcentos-7x-kmod-package-upgrade

but I got the following errors:

```
Dependencies Resolved

================================================================================
 Package           Arch          Version                       Repository  Size
================================================================================
Installing:
 zfs               x86_64        0.6.5.9-1.el7.centos          zfs        336 k
Installing for dependencies:
 libnvpair1        x86_64        0.6.5.9-1.el7.centos          zfs         38 k
 libuutil1         x86_64        0.6.5.9-1.el7.centos          zfs         43 k
 libzfs2           x86_64        0.6.5.9-1.el7.centos          zfs        125 k
 libzpool2         x86_64        0.6.5.9-1.el7.centos          zfs        425 k
 spl               x86_64        0.6.5.9-1.el7.centos          zfs         29 k
 spl-dkms          noarch        0.6.5.9-1.el7.centos          zfs        444 k
 zfs-dkms          noarch        0.6.5.9-1.el7.centos          zfs        1.9 M

Transaction Summary
================================================================================
Install  1 Package (+7 Dependent packages)

Total download size: 3.3 M
Installed size: 17 M
Is this ok [y/d/N]: y
Downloading packages:
libuutil1-0.6.5.9-1.el7.centos FAILED                                          
http://download.zfsonlinux.org/epel/7.4/x86_64/libuutil1-0.6.5.9-1.el7.centos.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
To address this issue please refer to the below knowledge base article 

https://access.redhat.com/articles/1320623

If above article doesn't help to resolve this issue please create a bug on https://bugs.centos.org/

libnvpair1-0.6.5.9-1.el7.cento FAILED                                          
http://download.zfsonlinux.org/epel/7.4/x86_64/libnvpair1-0.6.5.9-1.el7.centos.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
libzfs2-0.6.5.9-1.el7.centos.x FAILED                                          
http://download.zfsonlinux.org/epel/7.4/x86_64/libzfs2-0.6.5.9-1.el7.centos.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
libzpool2-0.6.5.9-1.el7.centos FAILED                                          
http://download.zfsonlinux.org/epel/7.4/x86_64/libzpool2-0.6.5.9-1.el7.centos.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
spl-0.6.5.9-1.el7.centos.x86_6 FAILED                                          
http://download.zfsonlinux.org/epel/7.4/x86_64/spl-0.6.5.9-1.el7.centos.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
spl-dkms-0.6.5.9-1.el7.centos. FAILED                                          
http://download.zfsonlinux.org/epel/7.4/x86_64/spl-dkms-0.6.5.9-1.el7.centos.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
zfs-0.6.5.9-1.el7.centos.x86_6 FAILED                                          
http://download.zfsonlinux.org/epel/7.4/x86_64/zfs-0.6.5.9-1.el7.centos.x86_64.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.
zfs-dkms-0.6.5.9-1.el7.centos. FAILED                                          
http://download.zfsonlinux.org/epel/7.4/x86_64/zfs-dkms-0.6.5.9-1.el7.centos.noarch.rpm: [Errno 14] HTTP Error 404 - Not Found
Trying other mirror.


Error downloading packages:
  libnvpair1-0.6.5.9-1.el7.centos.x86_64: [Errno 256] No more mirrors to try.
  spl-dkms-0.6.5.9-1.el7.centos.noarch: [Errno 256] No more mirrors to try.
  zfs-0.6.5.9-1.el7.centos.x86_64: [Errno 256] No more mirrors to try.
  libzpool2-0.6.5.9-1.el7.centos.x86_64: [Errno 256] No more mirrors to try.
  libuutil1-0.6.5.9-1.el7.centos.x86_64: [Errno 256] No more mirrors to try.
  spl-0.6.5.9-1.el7.centos.x86_64: [Errno 256] No more mirrors to try.
  zfs-dkms-0.6.5.9-1.el7.centos.noarch: [Errno 256] No more mirrors to try.
  libzfs2-0.6.5.9-1.el7.centos.x86_64: [Errno 256] No more mirrors to try.
```

I resolved that by `sudo yum autoremove`. So wiki should be updated as follows.

```
$ sudo yum remove zfs zfs-kmod spl spl-kmod libzfs2 libnvpair1 libuutil1 libzpool2 zfs-release
$ sudo yum install http://download.zfsonlinux.org/epel/zfs-release.el7_4.noarch.rpm
$ sudo yum autoremove
$ sudo yum install zfs 
```

**Comment by behlendorf** [comment]
Good point.  The wiki has been updated accordingly.

**Comment by lungati** [comment]
curl -I http://download.zfsonlinux.org/epel/7.4/x86_64/zfs-dkms-0.6.5.9-1.el7.centos.noarch.rpm
HTTP/1.1 404 Not Found
x-amz-request-id: F7F94A5CDF510F57
x-amz-id-2: /tzihp3K4MgkMxhcDngMFas3hG9nx5DMj9jdwkiquqKPm0zIVSL50x11N6AoPry1wydVyjtD2t8=
Content-Type: application/xml
Transfer-Encoding: chunked
Date: Fri, 17 Nov 2017 08:32:16 GMT
Server: AmazonS3

Are the packages still available?

**Comment by behlendorf** [comment]
@lungati CentOS 7.4 was released after zfs-0.6.5.9 was released and we don't go back and rebuild older versions to add to the repository.  The zfs-0.6.5.11 release and the zfs-0.7.x releases are available.

```
$ curl -I http://download.zfsonlinux.org/epel/7.4/x86_64/zfs-dkms-0.6.5.11-1.el7_4.noarch.rpm
HTTP/1.0 200 OK
x-amz-id-2: LMs2Slu0W9+Ta3tMbWm1RIG375FuBJxDga4FslalT8lBlCEyyjLOqiEQycxHQnntABF6ff2JQmY=
x-amz-request-id: E995B872C5ACD02F
Date: Fri, 17 Nov 2017 18:04:52 GMT
Last-Modified: Tue, 29 Aug 2017 21:10:50 GMT
ETag: "db524461097f8a2da515d653818a6efc"
Accept-Ranges: bytes
Content-Type: application/x-rpm
Content-Length: 1969740
Server: AmazonS3
X-Cache: MISS from eris
X-Cache-Lookup: MISS from eris:3128
Connection: close
```

**Comment by lungati** [comment]
yeah 0.7.3 worked and had to rollback to a previous kernel. thanks
